File: frontend/src/modules/graph/application/use-cases/readCommitFiles.ts
import { CommitFilesReader } from "@modules/graph/application/contracts/CommitFilesReader";
import { FileChange } from "@modules/graph/domain/fileChange";

export async function readCommitFiles(
  commitFilesReader: CommitFilesReader,
  repositoryPath: string,
  commitId: string
): Promise<FileChange[]> {
  return commitFilesReader.readCommitFiles(repositoryPath, commitId);
}

File: backend/src/app/commit_files/read_commit_files.rs
use std::path::Path;

use crate::app::commit_files::contracts::CommitFilesReader;
use crate::domain::errors::AppError;
use crate::domain::file_change::FileChange;

pub fn read_commit_files(
    commit_files_reader: &dyn CommitFilesReader,
    repository_path: &Path,
    commit_id: &str,
) -> Result<Vec<FileChange>, AppError> {
    commit_files_reader.read_commit_files(repository_path, commit_id)
}
